### PR TITLE
Erv2: publish inventory metrics

### PR DESCRIPTION
**Pull request by lechuk47**
Erv2: publish external resources inventory metrics 

APPSRE-10975

### CODE REVIEW

File: reconcile/utils/external_resources.py [hunk 1]
```diff
@@ -66,14 +65,14 @@ def managed_external_resources(namespace_info: Mapping[str, Any]) -> bool:
 
 
 def get_inventory_count_combinations(
-    inventory: ExternalResourceSpecInventory,
+    inventory: MutableMapping,
```
Comment (fishi0x01) [inline reconcile/utils/external_resources.py:68]: nit: why mutable is needed? It seems we do not expect any mutation here and `Mapping` might suffice for this case?

**Review by fishi0x01 — APPROVED** [review]

File: reconcile/utils/external_resources.py (hunk 1 above)
Comment (lechuk47) [reply reconcile/utils/external_resources.py]: You are right. I used the parent class of the two classes used with this method, but is not necessary to use Mutable in this case. Thanks for catching it.

**Review by cubismod — APPROVED** [review]

File: reconcile/utils/external_resources.py (hunk 1 above)
Comment (hemslo) [reply reconcile/utils/external_resources.py]: `Mapping` is too generic as we need to ensure key type match to use `(k.provision_provider, k.provisioner_name, k.provider) for k in inventory`, can we extract a common interface for it?

Comment (lechuk47) [reply reconcile/utils/external_resources.py]: I Switched it to a inventories static list.

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**